.github/workflows/build-depends.yml

```diff
@@ -51,23 +51,38 @@ jobs:
             depends/sources
           key: depends-sources-${{ hashFiles('depends/packages/*') }}
           restore-keys: |
-            depends-sources-${{ hashFiles('depends/packages/*') }}
             depends-sources-
 
+      - name: Cache SDKs
+        uses: actions/cache@v4
+        if: inputs.build-target == 'mac'
+        with:
+          path: |
+            depends/sdk-sources
+            depends/SDKs
+          key: depends-sdks-${{ hashFiles('depends/hosts/darwin.mk') }}
+          restore-keys: |
+            depends-sdks-
+
       - name: Restore cached depends
         uses: actions/cache/restore@v4
         id: restore
         with:
           path: |
             depends/built
             depends/${{ steps.setup.outputs.HOST }}
-          key: ${{ runner.os }}-depends-${{ inputs.build-target }}-${{ steps.setup.outputs.DEP_HASH }}-${{ hashFiles('depends/packages/*') }}
+          key: depends-${{ hashFiles('contrib/containers/ci/Dockerfile', 'depends/packages/*') }}-${{ inputs.build-target }}-${{ steps.setup.outputs.DEP_HASH }}
           restore-keys: |
-            ${{ runner.os }}-depends-${{ inputs.build-target }}-${{ hashFiles('depends/packages/*') }}
-            ${{ runner.os }}-depends-${{ inputs.build-target }}
+            depends-${{ hashFiles('contrib/containers/ci/Dockerfile', 'depends/packages/*') }}-${{ inputs.build-target }}-
+            depends-${{ hashFiles('contrib/containers/ci/Dockerfile', 'depends/packages/*') }}-
 
       - name: Build depends
-        run: env ${{ steps.setup.outputs.DEP_OPTS }} HOST=${{ steps.setup.outputs.HOST }} make -j$(nproc) -C depends
+        run: |
+          export HOST="${{ steps.setup.outputs.HOST  }}"
+          if [ "${HOST}" = "x86_64-apple-darwin" ]; then
+            ./contrib/containers/guix/scripts/setup-sdk
+          fi
```

Comment:
nit: feels wrong to use guix script in general ci

Reply:
`contrib/containers/guix` is a Dash-specific directory introduced in https://github.com/dashpay/dash/pull/5449 (Bitcoin's directory is `contrib/guix`) and `setup-sdk` was introduced in https://github.com/dashpay/dash/pull/6490, placed in that directory on review suggestion ([source](https://github.com/dashpay/dash/pull/6490#discussion_r1885800358)) but otherwise isn't tied to Guix at all.

We use it in GitLab CI without problems ([source](https://github.com/dashpay/dash/blob/bb469687d3d936f82fd8e8fbe0934eec5e17df5e/.gitlab-ci.yml#L50-L52)).